arangod/RestServer/Metric/MetricsFeature.cpp

```diff
@@ -0,0 +1,524 @@
+        "arangodb_scheduler_jobs_done",
+    };
+
+    v1suppressions = {
+        "arangodb_scheduler_jobs_dequeued_total",
+        "arangodb_scheduler_jobs_submitted_total",
+        "arangodb_scheduler_jobs_done_total",
+    };
+  } catch (std::exception const& e) {
+    LOG_TOPIC("efd51", ERR, Logger::MEMORY)
+        << "failed to allocate and populate the metrics v1/v2 translation "
+           "table "
+        << e.what();
+    FATAL_ERROR_EXIT();
+  }
+}
+
+void MetricsFeature::collectOptions(std::shared_ptr<options::ProgramOptions> options) {
+  _serverStatistics =
+      std::make_unique<ServerStatistics>(*this, StatisticsFeature::time());
+
+  options
+      ->addOption("--server.export-metrics-api", "turn metrics API on or off",
+                  new options::BooleanParameter(&_export),
+                  arangodb::options::makeDefaultFlags(arangodb::options::Flags::Hidden))
+      .setIntroducedIn(30600);
+
+  options
+      ->addOption("--server.export-read-write-metrics",
+                  "turn metrics for document read/write metrics on or off",
+                  new options::BooleanParameter(&_exportReadWriteMetrics),
+                  arangodb::options::makeDefaultFlags(arangodb::options::Flags::Hidden))
+      .setIntroducedIn(30707);
+}
+
+std::shared_ptr<Metric> MetricsFeature::doAdd(Builder& builder) {
+  auto metric = builder.build();
+  MetricKey key{metric->name(), metric->labels()};
+  bool success = false;
+  {
+    std::lock_guard guard(_lock);
+    success = _registry.try_emplace(key, metric).second;
+  }
+  if (!success) {
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, builder.type().data() +
+                                                           std::string{builder.name()} +
+                                                           " already exists");
+  }
+
+  return metric;
+}
+
+Metric* MetricsFeature::get(MetricKey const& key) {
+  std::lock_guard guard{_lock};
+  auto it = _registry.find(key);
+  if (it == _registry.end()) {
+    return nullptr;
+  }
+  return it->second.get();
+}
+
+bool MetricsFeature::remove(Builder const& builder) {
+  MetricKey key{builder.name(), builder.labels()};
+  std::lock_guard guard{_lock};
+  return _registry.erase(key) != 0;
+}
+
+bool MetricsFeature::exportAPI() const { return _export; }
+
+bool MetricsFeature::exportReadWriteMetrics() const {
+  return _exportReadWriteMetrics;
+}
+
+void MetricsFeature::validateOptions(std::shared_ptr<options::ProgramOptions>) {
+  if (_exportReadWriteMetrics) {
+    serverStatistics().setupDocumentMetrics();
+  }
+}
+
+void MetricsFeature::toPrometheus(std::string& result, bool v2) const {
+  // minimize reallocs
+  result.reserve(32768);
+
+  // QueryRegistryFeature
+  auto& q = server().getFeature<QueryRegistryFeature>();
+  q.updateMetrics();
+  {
+    bool changed = false;
+
+    std::lock_guard guard(_lock);
+    if (_globalLabels.find("shortname") == _globalLabels.end()) {
```

Comment:
with c++20 it can be simplified to
```suggestion
    if (!_globalLabels.contains("shortname")) {
```